MainDemo.Wpf/Fields.xaml

```diff
@@ -37,6 +37,63 @@
     </UserControl.Resources>
 
     <StackPanel Margin="8 0 16 16">
+
+        <TextBlock
+            Style="{StaticResource MaterialDesignHeadline5TextBlock}"
+            Text="'Reveal' style PasswordBoxes"/>
+        <Grid Margin="0,10,0,0">
+            <Grid.ColumnDefinitions>
+                <ColumnDefinition Width="Auto" />
+                <ColumnDefinition Width="20" />
+                <ColumnDefinition Width="Auto" />
+                <ColumnDefinition Width="20" />
+                <ColumnDefinition Width="Auto" />
+                <ColumnDefinition Width="20" />
+                <ColumnDefinition Width="Auto" />
+            </Grid.ColumnDefinitions>
+            <Grid.RowDefinitions>
+                <RowDefinition Height="30" />
+                <RowDefinition Height="70" />
+                <RowDefinition Height="80" />
+            </Grid.RowDefinitions>
+
+            <TextBlock Grid.Column="0" Grid.Row="0" Style="{StaticResource MaterialDesignBody2TextBlock}" VerticalAlignment="Top" Text="MaterialDesignRevealPasswordBox" />
+            <TextBlock Grid.Column="2" Grid.Row="0" Style="{StaticResource MaterialDesignBody2TextBlock}" VerticalAlignment="Top" Text="MaterialDesignFloatingHintRevealPasswordBox" />
+            <TextBlock Grid.Column="4" Grid.Row="0" Style="{StaticResource MaterialDesignBody2TextBlock}" VerticalAlignment="Top" Text="MaterialDesignFilledRevealPasswordBox" />
+            <TextBlock Grid.Column="6" Grid.Row="0" Style="{StaticResource MaterialDesignBody2TextBlock}" VerticalAlignment="Top" Text="MaterialDesignOutlinedRevealPasswordBox" />
+
+            <PasswordBox
```

Comment:
I do think pulling the password box into its own section would be a good idea. Also we need to wrap each of these in a `<smtx:XamlDisplay>` element so that they get the nice XAML preview.

Reply:
I moved the section down a bit and modified it slightly. It now uses ShowMeTheXAML wrappers and has an option for disabling the `PasswordBox`.

In doing the latter part, I actually found a bug. The opacity of the `PART_ClearButton` was not consistent across the styles due to double opacity being applied in certain cases. I fixed it so it is now consistent, and did the same for the "normal" `PasswordBox` styles and `TextBox` styles where the same bug was present.

The disabled states now look like this (with the clear button visible):
![image](https://user-images.githubusercontent.com/19572699/187520657-c8fc1034-493d-4420-b6d5-5e89e09db939.png)